### PR TITLE
MES-1982: Post-test Analytics

### PR DESCRIPTION
**Pull request by szabolcs-keresztes**
- [x] Adding analytics for screen tracking
- [x] Happy path post-test analytics
- [x] Errors post-test analytics

### CODE REVIEW

File: src/pages/office/office.ts [hunk 1]
```diff
@@ -542,6 +542,7 @@ export class OfficePage extends PracticeableBasePageComponent {
 
   defer() {
     this.popToRoot();
+    // this.store$.dispatch(new DeferWriteup());
```
Comment (ammarhaiderbjss) [inline src/pages/office/office.ts]: Do we not need this anymore?

File: src/modules/tests/tests.analytics.effects.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      const journalDataOfTest = test.journalData;
+
+      this.analytics.logEvent(
+        AnalyticsEventCategories.POST_TEST,
+        AnalyticsEvents.SUBMIT_TEST,
+        isTestPassed ? 'pass' : 'fail',
+      );
+
+      this.analytics.addCustomDimension(
+        AnalyticsDimensionIndices.TEST_ID, journalDataOfTest.testSlotAttributes.slotId.toString());
```
Comment (buzzub) [inline src/modules/tests/tests.analytics.effects.ts]: Could you use the `getCurrentTestSlotId` selector for this?

Comment (buzzub) [reply src/modules/tests/tests.analytics.effects.ts]: Ah maybe not... that selects `tests.currentTest.slotId`

Comment (buzzub) [reply src/modules/tests/tests.analytics.effects.ts]: Oh, you could just select `slotId` earlier and then reuse
`const slotId = getCurrentTestSlotId(tests);`

File: src/modules/tests/tests.selector.ts [hunk 3]
```diff
@@ -44,7 +48,7 @@ export const isTestOutcomeSet = (test: StandardCarTestCATBSchema) => {
   return false;
 };
 
-export const isPassed = (test: StandardCarTestCATBSchema) => {
+export const isPassed = (test: StandardCarTestCATBSchema): boolean => {
```
Comment (buzzub) [inline src/modules/tests/tests.selector.ts]: Niiice

File: src/pages/back-to-office/back-to-office.analytics.effects.ts [hunk 4]
```diff
@@ -32,4 +38,34 @@ export class BackToOfficeAnalyticsEffects {
...
+        select(getCurrentTest),
+        select(isPassed),
+      ),
+      this.store$.pipe(
+        select(getTests),
+        select(getCurrentTest),
+        select(getJournalData),
+      ),
+    ),
+    switchMap(([action, isPassed, journalDataOfTest]) => {
```
Comment (buzzub) [inline src/pages/back-to-office/back-to-office.analytics.effects.ts]: Is it worth specifying the type of these arguments? e.g.
`switchMap(([action, isPassed, journalDataOfTest]: [DeferWriteUp, boolean, JournalData]) => {`

File: src/modules/tests/tests.analytics.effects.ts (hunk 2 above)
Comment (szabolcs-keresztes) [reply src/modules/tests/tests.analytics.effects.ts]: The only thing with the `getCurrentTestSlotId` function is that it always gets the current test slot id. And there is a chance here that we are not setting the status of the current test, but an older test which failed to get submitted earlier. 
And the most easiest way is to go check what's in `journalData` and set that `slotId`.

File: src/pages/back-to-office/back-to-office.analytics.effects.ts (hunk 4 above)
Comment (szabolcs-keresztes) [reply src/pages/back-to-office/back-to-office.analytics.effects.ts]: Can do

File: src/pages/office/office.ts (hunk 1 above)
Comment (szabolcs-keresztes) [reply src/pages/office/office.ts]: changed it

**Pull request merged**